.github/workflows/windows.yml

```diff
@@ -17,87 +17,128 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  mingw:
-    runs-on: windows-2022
-    strategy:
-      matrix:
-        architecture: [x64, x86]
+  # 1) Simple MSVC build on latest image (Release)
+  msvc-latest:
+    runs-on: windows-latest
+    steps:
+      - uses: egor-tensin/setup-mingw@84c781b557efd538dec66bde06988d81cd3138cf
+        with:
+           version: 13.2.0
+           static: 0   # v4.2.2 (keep the hash!)
+       # --- missing checkout step ---
+      - name: Checkout repository                       # <-- this is what Niels meant
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2 pinned
 
+    # now the sources are present, so CMake can see them
+      - name: Configure CMake
+        run: cmake -S . -B build -G "MinGW Makefiles" -DJSON_BuildTests=On     
+      - name: Configure CMake
+        run: >
+          cmake -S . -B build -G "Visual Studio 17 2022"
```

Comment:
You previously ran CMake with "MinGW Makefiles" generator, now with "Visual Studio 17 2022". That does not work.